charts/catalog/templates/controller-manager-deployment.yaml

```diff
@@ -14,6 +14,8 @@ spec:
       app: {{ template "fullname" . }}-controller-manager
   template:
     metadata:
+      annotations:
+        prometheus.io/scrape: "{{ .Values.controllerManager.enablePrometheusScrape }}"
```

Comment:
Is this annotation key used elsewhere for this kind of thing?

Reply:
What exactly does this drive?  It's not clear to me whether this actually changes the behavior of the controller or whether this is meant to make prometheus aware that it has to scrape.

Reply:
Yes, this is a standard annotation to indicate to Prometheus that this pod exposes metrics.  Prometheus is configured with scrape configs which determine what endpoints should be polled to collect metrics from.  Prometheus provides a standard Kubernetes configuration that looks for any pods deployed with this metadata annotation - see 
https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml#L239-L273

This annotation is just advertising the fact that Prometheus metrics can be collected at the default /metrics endpoint.  You must still deploy and configure Prometheus for any metrics to be collected.